### PR TITLE
chore: improve cucumber tags and run time speed

### PR DESCRIPTION
**Pull request by SWvheerden**
Description
---

This PR improves the run time of certain tests to remove `waits` and make sure they have async functions to retest waitable events.
It adds a `@base-node` tag to all base node tests.
It adds a `@wallet` tag to all wallet tests.
It adds `@`  tags to all feature files.
Its adds and remove `@critical` tags to all required tests, adding where critical functionality is not tested and removing where it is double tested and or not critical to core functionality.

Motivation and Context
---
Made sure we can run all tests as short as possible. Tests are grouped with the following tags:
`@base-node` which can be used to run tests if verification of code base node functionality is desired
`@wallet` which can be used to run tests if verification of code wallet functionality is desired
`@critical` if a short test of all core functionality is desired.
`@long-running` if really detailed long-running tests are desired. Can take hours. 

Tests without a `@critical` and `@long-running` are tests that are required for passing CI, but will take longer to run.
Running test with `--tags @base-node AND @critical` or `--tags @wallet AND @critical` completes it run time below 10 minutes.

### CODE REVIEW

File: integration_tests/features/WalletQuery.feature [hunk 1]
```diff
@@ -1,13 +1,14 @@
-@wallet
+@wallet-query
```
Comment (stringhandler) [inline integration_tests/features/WalletQuery.feature]:
```suggestion
@wallet-query @wallet
```

File: integration_tests/features/WalletMonitoring.feature [hunk 2]
```diff
@@ -51,7 +48,7 @@ Feature: Wallet Monitoring
     Then all COINBASE transactions for wallet WALLET_A1 and wallet WALLET_B1 have consistent but opposing validity
 
   # 18+ mins on circle ci
-  @long-running
+  @long-running @wallet
```
Comment (stringhandler) [inline integration_tests/features/WalletMonitoring.feature]: You can just put this tag at the top of the file

**Review by philipr-za — COMMENTED** [review]
The CI run of the integration tests looks pretty broken?

**Comment by delta1** [comment]
Not sure what’s gone wrong with the grpc client now

File: integration_tests/features/WalletMonitoring.feature (hunk 2 above)
Comment (SWvheerden) [reply integration_tests/features/WalletMonitoring.feature]: Moved all '@wallet' and all '@base-node' tags where I could to the top

**Review by philipr-za — APPROVED** [review]
LGTM

**Pull request merged**